docs/pages/setup/guides/teleport-operator.mdx

```diff
@@ -0,0 +1,194 @@
+
+```code
+$ helm install teleport-cluster teleport/teleport-cluster \
+        --create-namespace --namespace teleport-cluster \
+        --set clusterName=teleport-cluster.teleport-cluster.svc.cluster.local \
+        --set operator.enabled=true \
+        --set teleportVersionOverride=(=teleport.version=)
+```
+
+This command installs the required Kubernetes CRDs and deploys the operator next to the Teleport cluster.
+All resources except the CRDs are created in the `teleport-cluster` namespace.
+
+All subsequents commands will take place in the `teleport-cluster` namespace, so let's use it by default:
+```code
+$ kubectl config set-context --current --namespace teleport-cluster
+```
+
+## Step 3/3. Manage users and roles using `kubectl`
+
+Create the manifest `teleport-resources.yaml` describing two Custom Resources: a `TeleportUser` and a `TeleportRole`:
+
+```yaml
+apiVersion: resources.teleport.dev/v5
+kind: TeleportRole
+metadata:
+  name: myrole
+spec:
+  allow:
+    rules:
+      - resources: ['user', 'role']
+        verbs: ['list','create','read','update','delete']
+---
+apiVersion: resources.teleport.dev/v2
+kind: TeleportUser
+metadata:
+  name: myuser
+spec:
+  roles: ['myrole']
+```
+
+Apply the manifests to the Kubernetes cluster:
+
+```code
+$ kubectl apply -f teleport-resources.yaml
+```
+
+List the created Kubernetes resources:
+
+```code
+$ kubectl get teleportroles
+# NAME     AGE
+# myrole   10m
+
+$ kubectl get teleportusers
+# NAME     AGE
+# myuser   10m
+```
+
+Check the user `myuser` has been created in Teleport and has been granted the role `myrole`:
+```code
+$ PROXY_POD=$(kubectl get po -l app=teleport-cluster -o jsonpath='{.items[0].metadata.name}')
+$ kubectl exec -it "$PROXY_POD" -c teleport -- tctl users ls
+# User                          Roles
+# ----------------------------- -----------------------------
+# bot-teleport-operator-sidecar bot-teleport-operator-sidecar
+# myuser                        myrole
+```
+
+At this point the Teleport Kubernetes Operator is functional and Teleport users and roles can be managed from
+Kubernetes.
+
+## Going further
+
+### Exploring Teleport Kubernetes CRDs
+
+Available fields can be browsed with `kubectl explain` in a cluster with Teleport CRDs installed.
+For example the command:
+```code
+$ kubectl explain teleportroles.spec
+```
+Returns the following fields:
+```shell
+KIND:     TeleportRole
+VERSION:  resources.teleport.dev/v5
+
+RESOURCE: spec <Object>
+
+DESCRIPTION:
+    Role resource definition v5 from Teleport
+
+FIELDS:
+   allow	<Object>
+     Allow is the set of conditions evaluated to grant access.
+
+   deny	<Object>
+     Deny is the set of conditions evaluated to deny access. Deny takes priority
+     over allow.
+
+   options	<Object>
+     Options is for OpenSSH options like agent forwarding.
+```
+
+### Troubleshooting a resource not reconciled in Teleport
+
+During reconciliation the operator adds a `status` field on the Kubernetes resource.
```

Comment:
I think we should explain reconciliation and the overall lifecycle of resources managed by the Operator. Right now, this is the first time the guide mentions reconciliation, and not all users may be familiar with the operator pattern.